library_registry/library_registration_protocol/controller.py

```diff
@@ -0,0 +1,346 @@
+import flask
+import json
+from smtplib import SMTPException
+from Crypto.PublicKey import RSA
+from Crypto.Cipher import PKCS1_OAEP
+from flask import (Response, render_template_string, request)
+from flask_babel import lazy_gettext as _
+from library_registry.util.string_helpers import (base64, random_string)
+from library_registry.model_helpers import (get_one, get_one_or_create)
+from library_registry.emailer import Emailer
+from library_registry.config import (Configuration, CannotLoadConfiguration, CannotSendEmail)
+from library_registry.opds import OPDSCatalog
+from library_registry.util.problem_detail import ProblemDetail
+from library_registry.util.shared_controller import BaseController, LibraryRegistryAnnotator
+from library_registry.library_registration_protocol.registrar import LibraryRegistrar
+from library_registry.admin.templates.templates import admin as admin_template
+from library_registry.util.http import HTTP
+from library_registry.constants import (
+    OPDS_CATALOG_REGISTRATION_MEDIA_TYPE,
+)
+from library_registry.problem_details import (
+    AUTHENTICATION_FAILURE,
+    INTEGRATION_ERROR,
+    NO_AUTH_URL,
+    UNABLE_TO_NOTIFY,
+)
+from library_registry.model import (
+    ConfigurationSetting,
+    Hyperlink,
+    Library,
+    Resource,
+    Validation,
+)
+
+class LibraryRegistryController(BaseController):
+
+    OPENSEARCH_TEMPLATE = (
+        '<?xml version="1.0" encoding="UTF-8"?>'
+        '<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">'
+        '<ShortName>%(name)s</ShortName>'
+        '<Description>%(description)s</Description>'
+        '<Tags>%(tags)s</Tags>'
+        '<Url type="application/atom+xml;profile=opds-catalog" template="%(url_template)s"/>'
+        '</OpenSearchDescription>'
+    )
```

Comment:
This is duplicate code that shouldn't be necessary in this class.

Reply:
Thank you. Removed.